src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -279,7 +279,7 @@ protected async Task<SyntaxNode> TryGetSelectedNodeAsync(Document document, Text
                     }
 
                     leftNode = leftNode.Parent;
-                    if (leftNode == null || leftNode.Span.End != location)
+                    if (leftNode == null || leftNode.GetLastToken().Span.End != location)
```

Comment:
tagging @petrroll .  Had to do this for the walking up logic to properly walk up the right side of a node that was missing tokens.  i.e. i want this feature to work even if the user does not have the semicolon at the end of an expression.  

I believe this is the logic we want as we want to walk up the right side of what the user actually has in source, not the arbitrary 'end' point the parser produced based on the invisible missing tokens it adds to the tree.

Reply:
Good catch! Didn't know parser adds invisible yet non-zero width tokens.